allowlists/templatemaker.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2011-2023 Lawrence Livermore National Security, LLC and other
+# msr-safe Project Developers. See the top-level COPYRIGHT file for
+# details.
+#
+# SPDX-License-Identifier: GPL-2.0-only
+
+import os
+import collections
+import regex as re
+
+scrap_dir = 'Intel_MSRs'
+
+class MSRMap:
+    def __init__(self, directory=scrap_dir):
+        self.msrs = {}
+        self.file_indices = {}
+        self.msr_names = {}
+        self.read_all_files(directory)
+        self.sorted = False
+
+    def add_msr(self, msr, fname, name):
+        self.sorted = False
+        # Treats (msr, arch) as unique (Ik Ik...)
+        if msr not in self.msrs:
+            self.msrs[msr] = []
+        self.msrs[msr].append(fname)
+
+        if msr not in self.msr_names:
+            self.msr_names[msr] = {}
+        self.msr_names[msr][fname] = name
+
+        # Inverted index
+        if fname not in self.file_indices:
+            self.file_indices[fname] = []
+        self.file_indices[fname].append(msr)
+
+
+    def get_categories(self, msr):
+        if not self.sorted:
+            self.sort()
+        return self.msrs.get(msr, [])
+    
+
+    def get_msrs(self, fname):
+        if not self.sorted:
+            self.sort()
+        return self.file_indices.get(fname, [])
+
+
+    def read_all_files(self, directory):
```

Comment:
This can be updated to not use REGEX now that it's using the blr data, but this works fine.